application/src/main/java/org/thingsboard/server/service/sync/ie/importing/impl/ResourceImportService.java

```diff
@@ -58,21 +58,26 @@ protected TbResource findExistingEntity(EntitiesImportCtx ctx, TbResource resour
     }
 
     @Override
-    protected boolean compare(EntitiesImportCtx ctx, EntityExportData<TbResource> exportData, TbResource prepared, TbResource existing) {
-        return true;
+    protected TbResource deepCopy(TbResource resource) {
+        return new TbResource(resource);
     }
 
     @Override
-    protected TbResource deepCopy(TbResource resource) {
-        return new TbResource(resource);
+    protected void cleanupForComparison(TbResource resource) {
+        super.cleanupForComparison(resource);
+        resource.setSearchText(null);
```

Comment:
Instead of redefining `equals`, here you may do the check for null node and set null if so.